src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentStack.cs

```diff
@@ -283,19 +294,12 @@ public void CopyTo(T[] array, int index)
         public void Push(T item)
         {
             // Pushes a node onto the front of the stack thread-safely. Internally, this simply
-            // swaps the current head pointer using a (thread safe) CAS operation to accomplish
-            // lock freedom. If the CAS fails, we add some back off to statistically decrease
-            // contention at the head, and then go back around and retry.
+            // swaps the current head pointer using a (thread safe) EXCH operation to accomplish
+            // lock freedom. Then it assings _next.
 
             Node newNode = new Node(item);
-            newNode._next = _head;
-            if (Interlocked.CompareExchange(ref _head, newNode, newNode._next) == newNode._next)
-            {
-                return;
-            }
-
-            // If we failed, go to the slow path and loop around until we succeed.
-            PushCore(newNode, newNode);
+            Node head = Interlocked.Exchange(ref _head, newNode);
+            newNode._next = head;
```

Comment:
Not concurrent-safe: there's a brief window where the queue would appear to be of length 1 (the current node).